src/main/java/io/github/syst3ms/skriptparser/lang/SimpleExpression.java

```diff
@@ -0,0 +1,60 @@
+package io.github.syst3ms.skriptparser.lang;
+
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.types.TypeManager;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Function;
+
+/**
+ * A simple expression with a fixed function to retrieve its values
+ * @param <T> the type of the values
+ */
+public class SimpleExpression<T> implements Expression<T> {
+    private final Class<? extends T> returnType;
+    private final boolean isSingle;
+    private final Function<TriggerContext, T[]> function;
+    @Nullable
+    private final String representation;
+
+    public SimpleExpression(Class<? extends T> returnType, boolean isSingle, Function<TriggerContext, T[]> function) {
+        this(returnType, isSingle, function, null);
+    }
+
+    public SimpleExpression(Class<? extends T> returnType, boolean isSingle, Function<TriggerContext, T[]> function, @Nullable String representation) {
+        this.returnType = returnType;
+        this.isSingle = isSingle;
+        this.function = function;
+        this.representation = representation;
+    }
+
+    @Override
+    @Contract("_, _, _ -> fail")
+    public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContext parseContext) {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
imo, it's better to remove this method and let IDEs and Java compiler throw an error

Reply:
I don't get what you mean by this. If the project can't compile, how can it work then?
This error is actually very descriptive and immediately tells you what you did wrong: you called a function which is not supported in the current syntax class.

Reply:
I mean than i think it will made an error when we compile skript-parser and not at the runtime, but i'm not sure